src/reducers/url-state.js

```diff
@@ -54,6 +54,8 @@ const dataSource: Reducer<DataSource> = (state = 'none', action) => {
       return 'unpublished';
     case 'SET_DATA_SOURCE':
       return action.dataSource;
+    case 'PROFILE_LOADED':
+      return action.dataSource ?? state;
```

Comment:
I'm not sure this is needed following the rest of the code. But I may be missing something?